File: ios/Plugin/DiagnosticPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(DiagnosticPlugin, "Diagnostic",
  CAP_PLUGIN_METHOD(requestAuthorization, CAPPluginReturnPromise);
  CAP_PLUGIN_METHOD(isFeatureAuthorized, CAPPluginReturnPromise);
  CAP_PLUGIN_METHOD(isGpsLocationEnabled, CAPPluginReturnPromise);
  CAP_PLUGIN_METHOD(getCurrentBatteryLevel, CAPPluginReturnPromise);
)

File: ios/Plugin/DiagnosticPlugin.swift
import Foundation
import Capacitor

@objc(DiagnosticPlugin)
public class DiagnosticPlugin: CAPPlugin {
    private let implementation = Diagnostic()

    @objc func requestAuthorization(_ call: CAPPluginCall) {
        implementation.requestAuthorization(call)
    }

    @objc func isFeatureAuthorized(_ call: CAPPluginCall) {
        implementation.isFeatureAuthorized(call)
    }

    @objc func isGpsLocationEnabled(_ call: CAPPluginCall) {
        implementation.isGpsLocationEnabled(call)
    }

    @objc func getCurrentBatteryLevel(_ call: CAPPluginCall) {
        implementation.getCurrentBatteryLevel(call)
    }
}
